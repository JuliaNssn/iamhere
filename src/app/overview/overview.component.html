<div class="container mt-5">
  <div
    class="d-flex justify-content-center"
    *ngIf="!userIsCheckedIn; else clock"
  >
    <button class="btn-check-in" mat-raised-button routerLink="/check-in">
      Check In
    </button>
  </div>
  <div class="d-flex mt-5 justify-content-center">
    <app-room
      class="m-2"
      *ngFor="let room of rooms$ | async"
      [name]="room.name"
      [checkedInUsers]="room.users"
    ></app-room>
  </div>
</div>

<ng-template #clock>
  <h1 class="text-center display-1">{{ now | date: "HH:mm" }}</h1>
</ng-template>
