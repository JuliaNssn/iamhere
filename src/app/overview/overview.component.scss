.btn-check-in {
  width: 200px;
  height: 48px;
  background: #6fb5bf;
  color: #fff;
}
