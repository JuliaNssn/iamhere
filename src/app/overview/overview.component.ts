import { Component, OnInit } from '@angular/core';

import { User } from '../shared/user/user.model';

@Component({
  selector: 'app-overview',
  templateUrl: './overview.component.html',
  styleUrls: ['./overview.component.scss']
})
export class OverviewComponent implements OnInit {
  users: User[];

  constructor() {}

  ngOnInit() {
    this.users = [
      {
        name: 'Jari Möllenbernd'
      },
      {
        name: 'Madita Schöner'
      },
      {
        name: 'Julia Nissen'
      },
      {
        name: 'Marco Klaassen'
      }
    ];
  }
}
