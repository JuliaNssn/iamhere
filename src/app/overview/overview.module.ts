import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { RouterModule } from '@angular/router';
import { OverviewComponent } from './overview.component';

@NgModule({
  declarations: [OverviewComponent],
  imports: [CommonModule, MatButtonModule, RouterModule],
  exports: [OverviewComponent]
})
export class OverviewModule {}
