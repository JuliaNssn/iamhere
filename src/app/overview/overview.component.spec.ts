import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { OverviewComponent } from './overview.component';
import { OverviewModule } from './overview.module';

describe('OverviewComponent', () => {
  let component: OverviewComponent;
  let fixture: ComponentFixture<OverviewComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [OverviewModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(OverviewComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
