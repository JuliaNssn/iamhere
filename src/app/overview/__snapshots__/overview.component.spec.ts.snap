// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewComponent should match the snapshot 1`] = `
<app-overview>
  <h1>
    Overview
  </h1>
</app-overview>
`;
